protocol/sendversion/sendversion.go

```diff
@@ -0,0 +1,62 @@
+package sendversion
+
+import (
+	"github.com/kaspanet/kaspad/blockdag"
+	"github.com/kaspanet/kaspad/config"
+	"github.com/kaspanet/kaspad/netadapter"
+	"github.com/kaspanet/kaspad/version"
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/pkg/errors"
+	"time"
+)
+
+var (
+	// userAgentName is the user agent name and is used to help identify
+	// ourselves to other kaspa peers.
+	userAgentName = "kaspad"
+
+	// userAgentVersion is the user agent version and is used to help
+	// identify ourselves to other kaspa peers.
+	userAgentVersion = version.Version()
+
+	// defaultServices describes the default services that are supported by
+	// the server.
+	defaultServices = wire.SFNodeNetwork | wire.SFNodeBloom | wire.SFNodeCF
+
+	// defaultRequiredServices describes the default services that are
+	// required to be supported by outbound peers.
+	defaultRequiredServices = wire.SFNodeNetwork
+)
+
+// SendVersion sends a version to a peer and waits for verack.
+func SendVersion(msgChan <-chan wire.Message, router *netadapter.Router, netAdapter *netadapter.NetAdapter,
+	dag *blockdag.BlockDAG) error {
+
+	selectedTipHash := dag.SelectedTipHash()
+	subnetworkID := config.ActiveConfig().SubnetworkID
+
+	// Version message.
+	msg := wire.NewMsgVersion(netAdapter.GetBestLocalAddress(), netAdapter.ID(), selectedTipHash, subnetworkID)
+	err := msg.AddUserAgent(userAgentName, userAgentVersion, config.ActiveConfig().UserAgentComments...)
```

Comment:
MsgVersion.AddUserAgent is called only when sending the user agent.
As such, why do we really want to validate it, and then panic if it fails? Why not just accept it as is 

(We can write a unit test to validate it, to make sure the default one is OK, and then on start we should validate it to make sure that a custom one is OK)

Reply:
You also validate it on Decode and Encode, but I don't care if it'll be validated only there